File: Iniciante/1021 - Notas e Moedas/Python/1021.py
def main():
    real_str, cents_str = input().split('.')
    value_in_cents = int(real_str) * 100 + int(cents_str)
    banknotes_in_cents = [10000, 5000, 2000, 1000, 500, 200]
    coins_in_cents = [100, 50, 25, 10, 5, 1]
    leftover = calculation_banknotes(value_in_cents, banknotes_in_cents)
    calculation_coins(leftover, coins_in_cents)


def print_out(text):
    print(text)


def calculation_banknotes(value, banknotes_in_cents):
    print_out("NOTAS:")
    for banknote in banknotes_in_cents:
        amount_banknotes = value // banknote
        print_out(f"{amount_banknotes} nota(s) de R$ {banknote/100:.2f}")
        value %= banknote
    return value


def calculation_coins(value, coins_in_cents):
    print_out("MOEDAS:")
    for coin in coins_in_cents:
        amount_coins = value // coin
        print_out(f"{amount_coins} moeda(s) de R$ {coin/100:.2f}")
        value %= coin


if __name__ == "__main__":
    main()

File: Iniciante/1021 - Notas e Moedas/JavaScript/1021.js
function main() {
  const input =  require('fs').readFileSync('/dev/stdin', 'utf8') ;
  let value = parseFloat(input);
  const banknotesList = [100, 50, 20, 10, 5, 2];
  const coinsList = [1.0, 0.5, 0.25, 0.1, 0.05, 0.01];
  const leftover = calculationBanknotes(value, banknotesList);
  calculationCoins(leftover, coinsList);
}

function print(text) {
  console.log(text);
}

function calculationBanknotes(value, banknotesList) {
  print("NOTAS:");
  for (let banknote of banknotesList) {
    let amountBanknotes = Math.trunc(value / banknote);
    print(`${amountBanknotes.toFixed(0)} nota(s) de R$ ${banknote}.00`);
    value -= amountBanknotes * banknote;
  }
  return value;
}

function calculationCoins(leftover, coinsList) {
  print("MOEDAS:");
  for (let coin of coinsList) {
    let amountCoins = Math.trunc((leftover + 0.00001) / coin);
    print(`${amountCoins.toFixed(0)} moeda(s) de R$ ${coin.toFixed(2)}`);
    leftover = parseFloat((leftover - amountCoins * coin).toFixed(2));
  }
}

main();
